### ISSUE TITLE
Documentation does not reflect a breaking change in pugsql 0.3.0

### ISSUE THREAD

**Issue by epgui** [mcfunley/pugsql#73]
The documentation includes an example almost identical to the following for multi-row inserts:

```sql
-- :name create_foo :insert
insert into foo (id, val) values (:id, :val)
```
```python
queries.create_foo(
  { 'id': 2, 'val': 'x' },
  { 'id': 3, 'val': 'y' },
  { 'id': 4, 'val': 'z' },
  { 'id': 5, 'val': 'a' },  # Not in original example
  { 'id': 6, 'val': 'b' },  # Not in original example
  { 'id': 7, 'val': 'c' },  # Not in original example
)
```

This used to work in pugsql 0.2.4. However, something seems to have changed in SQLAlchemy with pugsql 0.3.0 because now it gives an error similar to this:

```
self = <pugsql.compiler.Module object at 0x7f4d80417920>
clause = <sqlalchemy.sql.elements.TextClause object at 0x7f4d804178c0>
multiparams = ({'id': 2, 'val': 'x'}, {'id': 2, 'val': 'x'}, ...)
params = {}, conn = <sqlalchemy.engine.base.Connection object at 0x7f4d8027c230>
    def _execute(self, clause, *multiparams, **params):
        if getattr(self._locals, 'session', None):
            return self._locals.session.execute(clause, multiparams or params)
    
        if not self.engine:
            raise NoConnectionError()
    
        with self.engine.connect() as conn:
            if multiparams:
>               result = conn.execute(clause, *multiparams)
E               TypeError: Connection.execute() takes from 2 to 3 positional arguments but 8 were given
.venv/lib/python3.12/site-packages/pugsql/compiler.py:145: TypeError
```

However, it seems like passing the records in an array works:

```python
queries.create_foo(
  [
    { 'id': 2, 'val': 'x' },
    { 'id': 3, 'val': 'y' },
    { 'id': 4, 'val': 'z' },
    { 'id': 5, 'val': 'a' },
    { 'id': 6, 'val': 'b' },
    { 'id': 7, 'val': 'c' },
  ],
)
```

**Comment by mcfunley** [comment]
Thanks, yes 0.2 -> 0.3 upgraded SQLAlchemy to a modern version. I guess we didn't have a test case that covered this syntax. 

Breaking it might be necessary, I am not sure, but the error could probably be more helpful.

**Comment by epgui** [comment]
IMO this shouldn't really be a major problem... But the documentation should probably be updated.

PS-- thanks for the new release, this is a great little library! 🙏

**Comment by mcfunley** [comment]
Thank you that's nice of you to say!

Docs should be updated now. I'll leave this here to resolve the API usability at some point.

**Comment by epgui** [comment]
Oops, it looks like there's a complication:

```python
# This crashes
queries.create_foo(
  { 'id': 2, 'val': 'x' },
  { 'id': 3, 'val': 'y' },
  { 'id': 4, 'val': 'z' },
  { 'id': 5, 'val': 'a' },
  { 'id': 6, 'val': 'b' },
  { 'id': 7, 'val': 'c' },
)

# This crashes
with queries.transaction():
  queries.create_foo(
    [
      { 'id': 2, 'val': 'x' },
      { 'id': 3, 'val': 'y' },
      { 'id': 4, 'val': 'z' },
      { 'id': 5, 'val': 'a' },
      { 'id': 6, 'val': 'b' },
      { 'id': 7, 'val': 'c' },
    ],
  )

# This works
queries.create_foo(
  [
    { 'id': 2, 'val': 'x' },
    { 'id': 3, 'val': 'y' },
    { 'id': 4, 'val': 'z' },
    { 'id': 5, 'val': 'a' },
    { 'id': 6, 'val': 'b' },
    { 'id': 7, 'val': 'c' },
  ],
)

# This works
with queries.transaction():
  queries.create_foo(
    { 'id': 2, 'val': 'x' },
    { 'id': 3, 'val': 'y' },
    { 'id': 4, 'val': 'z' },
    { 'id': 5, 'val': 'a' },
    { 'id': 6, 'val': 'b' },
    { 'id': 7, 'val': 'c' },
  )
```

**Comment by mcfunley** [comment]
oh fascinating. yeah that definitely needs to be regularized somehow.

**Comment by epgui** [comment]
My knowledge of SqlAlchemy is not great, but I believe this is the relevant change: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#execute-method-more-strict-execution-options-are-more-prominent

Acknowledging that there may very well be cleaner or better approaches I'm just not seeing, I believe the lowest-effort fix may be to modify `pugsql.Module._execute()` as follows:

```python
from sqlalchemy.engine.base import Connection, Transaction

# ...

    def _execute(self, clause, *multiparams, **params):
        if getattr(self._locals, 'session', None):
            return self._locals.session.execute(clause, multiparams or params)

        if not self.engine:
            raise NoConnectionError()

        with self.engine.connect() as conn:
            if multiparams:
                if isinstance(conn, Connection):
                    result = conn.execute(clause, multiparams)
                elif isinstance(conn, Transaction):
                    result = conn.execute(clause, *multiparams)
                else:
                    raise TypeError  # Are there other cases?
            else:
                result = conn.execute(clause, params)
            conn.commit()
            return result
```

**Comment by mcfunley** [comment]
Should be fixed here:

https://github.com/mcfunley/pugsql/commit/4f4740dea3708683136334d82b3b866ae2aff743

I tried to make it consistently require the brackets, since that seems like the clearest option. Are you able to confirm this works against HEAD? Will release this as 0.3.1 next week I think.

**Comment by epgui** [comment]
I believe that will work. The new test is persuasive.

**Comment by mcfunley** [comment]
shipped that in 0.3.1
